### DISCUSSION TITLE
Caching?

### DISCUSSION THREAD

**Discussion by alecmev** [paularmstrong/onerepo#669] [Q&A] [answered]
I read [Why oneRepo > Alternatives & pitfalls > Cache inconsistency](https://onerepo.tools/concepts/why-onerepo/#cache-inconsistency). It explains what can go wrong when declaring inputs and outputs manually (as it's done in Nx and Turbo), but stops short of saying that there's a better way.

Do I infer correctly that oneRepo's philosophy then is "don't cache, just use source dependencies"? That sounds good for transpilation (e.g., TypeScript), but what's the story for code generation (e.g., [GraphQL Code Generator](https://the-guild.dev/graphql/codegen/docs/getting-started)), asset transformation (e.g., [gltfpack](https://github.com/zeux/meshoptimizer/tree/master/gltf)), etc.?

Also, I might be missing something, but one can disable all caching in Turbo and get similar results/guarantees, right? Not saying this to diminish oneRepo, just clarifying.

**Comment by paularmstrong** [comment, marked as answer]
> Do I infer correctly that oneRepo's philosophy then is "don't cache, just use source dependencies"? 

Yes, that's correct.

> but what's the story for code generation

The same. If you'd like to implement your own caching strategy for specific packages, you're always welcome to do so. For example, the Prettier and ESLint plugins use their built-in caching strategies by default.

> Also, I might be missing something, but one can disable all caching in Turbo and get similar results/guarantees, right? Not saying this to diminish oneRepo, just clarifying.

Turbo is essentially Lerna with caching. oneRepo is an entire integrated CLI SDK that enables you to build on top of the Workspace Graph for custom workflows.
